src/ORM/Marshaller.php

```diff
@@ -216,7 +216,8 @@ protected function _prepareDataAndOptions($data, $options)
 
         $tableName = $this->_table->alias();
         if (isset($data[$tableName])) {
-            $data = $data[$tableName];
+            $data += $data[$tableName];
+            unset($data[$tableName]);
```

Comment:
I think that difference would be present even if the `unset()` wasn't there. But you are right that different data will be present now than in previous versions. I think this isn't a common case though. Originally the prefix was used so that people could have an easier time migrating their views from 2.x to 3.x. 

```php
$this->input('Articles.title')`
```

Wouldn't need to be updated to work in 3.x, when migrating from 2.x. It was not common in 2.x to do `Articles.Comments.comment`